src/index.ts

```diff
@@ -0,0 +1,33 @@
+const fs = require('fs');
+const octokit = require('@octokit/rest')();
+const createGithubService = require('./createGithubService').default;
+const deploy = require('./deploy').default;
+const exec = require('child_process').exec;
+
+async function sgAutorelease({rootPath, sourceDirectory, domain, repoOwner, repoName, githubToken, pr}) {
+  const githubServiceParams = {
+    owner: repoOwner,
+    repo: repoName,
+    token: githubToken
+  };
+  const valueExists = (value): boolean => !!value;
+  const githubServiceParamsAreValid = Object.values(githubServiceParams).every(valueExists);
+  const githubService = githubServiceParamsAreValid && createGithubService(octokit, githubServiceParams);
+  const fileService = {
+    exists: path => fs.existsSync(path)
+  };
+  const surgeService = command => new Promise((resolve, reject) => {
+    exec(`npx surge ${command.join(' ')}`, {timeout: 10000}, (error, stdout, stderr) => {
```

Comment:
so I assume the node version of it could not be easily used

Reply:
Turned out to be `cli` oriented. The use of `exec` instead of `spawn` does compact it greatly and `timeout` is key because in case it get invalid data it will halt and expect user input (which is why it hung)